src/main/kotlin/lotto/controller/LottoController.kt

```diff
@@ -0,0 +1,68 @@
+package lotto.controller
+
+import lotto.model.*
+import lotto.service.InputService
+import lotto.service.RankService
+import lotto.view.OutputView
+
+class LottoController {
+
+    private val outputView = OutputView()
+    private val lottoGenerator = LottoGenerator()
+
+    fun run() {
+        val purchaseAmount = getPurchaseAmount()
+        val purchaseQuantity = purchaseAmount.calculatePurchaseQuantity()
+        val lottoTickets = getLottoTickets(purchaseQuantity)
+        outputView.printPurchaseLotto(lottoTickets, purchaseQuantity)
+
+        val winningNumbers = getWinningNumbers()
+        val winningLotto = getWinningLotto(winningNumbers)
+        val rankCount = RankService.calculateLottoRank(winningLotto.winningNumbers, winningLotto.bonusNumber, lottoTickets)
+
+        outputView.printMatchingResult(rankCount)
+        val totalReturn = RankService.calculateTotalReturn(purchaseAmount)
+        outputView.printTotalRateOfReturn(totalReturn)
+    }
+
+    private fun getPurchaseAmount(): PurchaseAmount {
+        return try {
+            outputView.printPurchaseAmountRequest()
+            val userInput = InputService.readLineNumber()
+            PurchaseAmount(userInput)
+        } catch(e:IllegalArgumentException) {
+            println(e.message)
+            getPurchaseAmount()
+        }
+    }
```

Comment:
return try와 재귀함수를 사용하는 방법도 있군요.. 배워갑니다! 추가로 궁굼한점인데 return try를 사용하면 식이니까 아래와 같이 단일 표현식으로도 가능할거 같은데 혹시 적용이 될까요??
```suggestion
    private fun getPurchaseAmount(): PurchaseAmount =
        return try {
            outputView.printPurchaseAmountRequest()
            val userInput = InputService.readLineNumber()
            PurchaseAmount(userInput)
        } catch(e:IllegalArgumentException) {
            println(e.message)
            getPurchaseAmount()
        }
```